_search-plugins/concurrent-segment-search.md

```diff
@@ -117,8 +196,18 @@ Non-concurrent search calculates the document count error and returns it in the
 For more information about how `shard_size` can affect both `doc_count_error_upper_bound` and collected buckets, see [this GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/11680#issuecomment-1885882985).
 
 
-## Developer information: AggregatorFactory changes
+## Developer information: 
+
+### AggregatorFactory changes
 
 Because of implementation details, not all aggregator types can support concurrent segment search. To accommodate this, we have introduced a [`supportsConcurrentSegmentSearch()`](https://github.com/opensearch-project/OpenSearch/blob/bb38ed4836496ac70258c2472668325a012ea3ed/server/src/main/java/org/opensearch/search/aggregations/AggregatorFactory.java#L121) method in the `AggregatorFactory` class to indicate whether a given aggregation type supports concurrent segment search. By default, this method returns `false`. Any aggregator that needs to support concurrent segment search must override this method in its own factory implementation. 
 
 To ensure that a custom plugin-based `Aggregator` implementation works with the concurrent search path, plugin developers can verify their implementation with concurrent search enabled and then update the plugin to override the [`supportsConcurrentSegmentSearch()`](https://github.com/opensearch-project/OpenSearch/blob/bb38ed4836496ac70258c2472668325a012ea3ed/server/src/main/java/org/opensearch/search/aggregations/AggregatorFactory.java#L121) method to return `true`.
+
+### Pluggable concurrent search deciders: ConcurrentSearchRequestDecider
+
+Plugin developers can customize the concurrent search decision-making for `auto` mode by implementing the [`ConcurrentSearchRequestDecider`](https://github.com/opensearch-project/OpenSearch/blob/c505709d73587fd2fdb714d24510b44ae07aa4a6/server/src/main/java/org/opensearch/search/deciders/ConcurrentSearchRequestDecider.java#L29).
```

Comment:
```suggestion
Plugin developers can customize the concurrent search decision-making for `auto` mode by implementing the [`ConcurrentSearchRequestDecider`](https://github.com/opensearch-project/OpenSearch/blob/2.x/server/src/main/java/org/opensearch/search/deciders/ConcurrentSearchRequestDecider.java).
```

Let's just link against `2.x` so we don't have to change this if there are any changes in the future.

Reply:
Same for the below 2 links too